### PR TITLE
GIDS: Upgrade to Ruby v2.6.6 #5747

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/5747

Ruby 2.6.6 includes security fixes.  [Release notes](https://www.ruby-lang.org/en/news/2020/03/31/ruby-2-6-6-released/)

- When doing `rbenv install 2.6.6` had to do `brew update && brew upgrade ruby-build` first
- Due to `ArgumentError: You tried to define a scope named "filter" on the model "InstitutionProgram", but ActiveRecord::Relation already defined an instance method with the same name.` needed rename two scopes from `filter` to `filter_results`

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/models/institution.rb [hunk 1]
```diff
@@ -252,7 +252,7 @@ def self.autocomplete(search_term, limit = 6)
     )
   }
 
-  scope :filter, lambda { |field, value|
+  scope :filter_result, lambda { |field, value|
```
Comment (zurbergram) [inline app/models/institution.rb:255]:
```
ArgumentError: You tried to define a scope named "filter" on the model "InstitutionProgram", but ActiveRecord::Relation already defined an instance method with the same name.
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/institution_program.rb:67:in `<class:InstitutionProgram>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/institution_program.rb:3:in `<top (required)>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/program.rb:18:in `<class:Program>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/program.rb:3:in `<top (required)>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/config/initializers/csv_types.rb:31:in `<top (required)>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/config/environment.rb:5:in `<top (required)>'
/Users/matthewroth/.rbenv/versions/2.6.6/bin/bundle:23:in `load'
/Users/matthewroth/.rbenv/versions/2.6.6/bin/bundle:23:in `<main>'
```

File: app/models/institution_program.rb [hunk 2]
```diff
@@ -64,7 +64,7 @@ def va_bah
     )
   }
 
-  scope :filter, lambda { |field, value|
+  scope :filter_result, lambda { |field, value|
```
Comment (zurbergram) [inline app/models/institution_program.rb:67]:
```
ArgumentError: You tried to define a scope named "filter" on the model "InstitutionProgram", but ActiveRecord::Relation already defined an instance method with the same name.
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/institution_program.rb:67:in `<class:InstitutionProgram>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/institution_program.rb:3:in `<top (required)>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/program.rb:18:in `<class:Program>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/app/models/program.rb:3:in `<top (required)>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/config/initializers/csv_types.rb:31:in `<top (required)>'
/Users/matthewroth/dev/vets-gov/gibct-data-service/config/environment.rb:5:in `<top (required)>'
/Users/matthewroth/.rbenv/versions/2.6.6/bin/bundle:23:in `load'
/Users/matthewroth/.rbenv/versions/2.6.6/bin/bundle:23:in `<main>'
```

File: app/models/institution.rb (hunk 1 above)
Comment (zurbergram) [inline app/models/institution.rb:255]: changed name since another defined scope with the name `filter`, same issue as InstitutionPrograms

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: app/models/institution.rb (hunk 1 above)
Comment (LindseySaari) [reply app/models/institution.rb:255]: Nice ambiguity catch 👍

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**